### PR TITLE
fix: UPE note when account disconnected

### PR DESCRIPTION
**Pull request by frosso**
# Changes proposed in this Pull Request:
Let's not display the UPE inbox note if the Stripe account is not connected.

# Testing instructions
- Ensure your account is connected (keys for should be entered in order to be in that state)
- Ensure the Stripe gateway is enabled
- Ensure you have the `_wcstripe_feature_upe_settings` flag enabled
- The note should be displayed

### CODE REVIEW

**Comment by harriswong** [comment]
👀

File: tests/phpunit/test-class-wc-stripe-notes.php [hunk 1]
```diff
@@ -102,3 +115,7 @@ public function test_create_upe_availability_note_does_not_create_note_when_upe_
 		$this->assertSame( 0, count( $admin_note_store->get_notes_with_name( WC_Stripe_UPE_Availability_Note::NOTE_NAME ) ) );
 	}
 }
+
+//function woocommerce_gateway_stripe() {
+//	return WC_Stripe_Inbox_Notes_Test::$global_woocommerce_gateway_stripe;
+//}
```
Comment (harriswong) [inline tests/phpunit/test-class-wc-stripe-notes.php]: Do we still need these? :D

Comment (frosso) [reply tests/phpunit/test-class-wc-stripe-notes.php]: haha woops - I was trying different testing approaches :D Thanks! fixed in aa758710

File: tests/phpunit/test-class-wc-stripe-notes.php [hunk 2]
```diff
@@ -9,9 +9,21 @@
...
+
 	public function setUp() {
 		parent::setUp();
 
+		// overriding the `WC_Stripe_Connect` in woocommerce_gateway_stripe(),
+		// because the method we're calling is static and we don't really have a way of injecting it all the way down to this class.
+		$this->stripe_connect_mock = $this->createPartialMock( WC_Stripe_Connect::class, [ 'is_connected' ] );
+		$this->stripe_connect_mock->expects( $this->any() )->method( 'is_connected' )->willReturn( true );
+		$this->stripe_connect_original        = woocommerce_gateway_stripe()->connect;
+		woocommerce_gateway_stripe()->connect = $this->stripe_connect_mock;
```
Comment (harriswong) [inline tests/phpunit/test-class-wc-stripe-notes.php:25]: Would be nice if `woocommerce_gateway_stripe()` is `woocommerce_gateway_stripe( $instance )`. Then we can inject the mock into it.

**Review by harriswong — APPROVED** [review]
- Tested removing existing note. Then refresh page without Stripe key, verified no note.
- Added Stripe key, refresh page, verified note shows.

👍

File: tests/phpunit/test-class-wc-stripe-notes.php (hunk 2 above)
Comment (frosso) [reply tests/phpunit/test-class-wc-stripe-notes.php:25]: Yeah - or if we even had a way of injecting it into the note class :(

**Pull request merged**